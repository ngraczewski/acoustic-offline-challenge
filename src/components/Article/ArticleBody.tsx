import React from 'react';

export type Props = {
  body: string[];
};

export const ArticleBody = ({ body }: Props): JSX.Element => {
  return (
    <div className="article__body">
      {body.map((b, index) => (
        // since we are not actively modifying this list, using index as a key
        // is sufficent, in other cases something more coupled with the item
        // would be required
        <div key={index} dangerouslySetInnerHTML={{ __html: b }} />
      ))}
    </div>
  );
};
