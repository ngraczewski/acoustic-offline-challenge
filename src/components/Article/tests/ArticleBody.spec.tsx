import React from 'react';
import faker from 'faker';
import { shallow } from 'enzyme';
import { ArticleBody, Props } from '../ArticleBody';

describe('ArticleBody', () => {
  beforeEach(() => {
    faker.seed(1);
  });

  it('should render correctly', () => {
    const props: Props = {
      body: [
        `<p>${faker.lorem.paragraph()}</p>`,
        `<p>${faker.lorem.paragraph()}</p>`,
      ],
    };
    const wrapper = shallow(<ArticleBody {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
