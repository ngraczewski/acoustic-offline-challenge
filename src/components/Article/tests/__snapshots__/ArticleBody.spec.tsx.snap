// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleBody should render correctly 1`] = `
<div
  className="article__body"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Quisquam recusandae alias consequuntur corporis repellat ratione ut sunt qui. Iure ut libero qui. Ut nulla quam ipsam nobis cupiditate sed dignissimos debitis incidunt. Sed libero repudiandae.</p>",
      }
    }
    key="0"
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Natus et eos itaque velit omnis. Porro ut et ipsam explicabo eligendi occaecati debitis et. Eum dicta eum eaque enim ipsum inventore debitis libero aspernatur. Quam tempore a velit provident velit eligendi.</p>",
      }
    }
    key="1"
  />
</div>
`;
